telegram/_bot.py

```diff
@@ -193,8 +229,34 @@ class Bot(TelegramObject, contextlib.AbstractAsyncContextManager["Bot"]):
 
     Args:
         token (:obj:`str`): Bot's unique authentication token.
-        base_url (:obj:`str`, optional): Telegram Bot API service URL.
+        base_url (:obj:`str` | Callable[[:obj:`str`], :obj:`str`], optional): Telegram Bot API
```

Comment:
I think we should also put an Example: here to show how to use it. Easy to get confused and put your full token in there instead of the placeholder. And/Or maybe a Caution: to say to put the placeholder and not the actual token.

Reply:
IMO a caution is a bit too much. I'll add examples for the test env though :thumbsup: